lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -8336,14 +8329,6 @@ namespace Js
         if (sharedPropertyGuards != nullptr)
         {
             auto guards = AnewArray(alloc, Js::PropertyId, sharedPropertyGuards->Count());
-
-            struct AutoResizeLock
-            {
-                AutoResizeLock(SharedPropertyGuardDictionary* d) :d(d) { d->LockResize(); }
-                ~AutoResizeLock() { d->UnlockResize(); }
-                SharedPropertyGuardDictionary* d;
-            } autoResizeLock(sharedPropertyGuards);
```

Comment:
just took a closer look, we might still need lock on both location. not for old buffer being deleted, but for the count and the iterator to be consistent. while adding to the dictionary, it increase the count first, and then write the buffer, if background thread finished a get operation in between, it'll get an count-1 entries